### DISCUSSION TITLE
Nextcloud AIO and collabora with Sophos XG and Cloudflare reverse proxies.

### DISCUSSION THREAD

**Discussion by panoskpv** [nextcloud/all-in-one#4203] [Questions] [answered]
I have installed Nextcloud AIO and i have a local Sophos XG reverse proxy and Cloudflare. As far as I understand how sophos xg reverse proxy works I need to enable websockets to specific paths. Which should be this path(s)?  I keep getting "Failed to establish socket connection or socket connection closed unexpectedly." error. I have tried adding 0.0.0.0 method for WOPI and enabling werbsockets to "/apps/richdocumentscode/proxy.php?req=/" and "/apps/richdocumentscode/" with no luck.

**Comment by szaimen** [comment, marked as answer]
Hi, so the best way would indeed be to enable websockets for the whole domain. If that is actually not possible, you might have a look at https://github.com/nextcloud/all-in-one/blob/main/Containers/apache/Caddyfile
